Profiler build error on musl libc because of `stat64`

On musl libc 1.2.5, building `tracy-profiler` results in an error because of `stat64`:
```sh
In file included from /home/willian/dev/software/tracy/server/TracyWorker.cpp:34:
/home/willian/dev/software/tracy/server/TracyFileRead.hpp: In constructor 'tracy::FileRead::FileRead(FILE*, const char*)':
/home/willian/dev/software/tracy/server/TracyFileRead.hpp:461:23: error: aggregate 'tracy::FileRead::FileRead(FILE*, const char*)::stat64 buf' has incomplete type and cannot be defined
  461 |         struct stat64 buf;
      |                       ^~~
/home/willian/dev/software/tracy/server/TracyFileRead.hpp:462:30: error: invalid use of incomplete type 'struct tracy::FileRead::FileRead(FILE*, const char*)::stat64'
  462 |         if( stat64( fn, &buf ) == 0 )
      |                              ^
/home/willian/dev/software/tracy/server/TracyFileRead.hpp:461:16: note: forward declaration of 'struct tracy::FileRead::FileRead(FILE*, const char*)::stat64'
  461 |         struct stat64 buf;
      |                ^~~~~~
```

This happens because musl does not have the `stat64` type exposed, at least not without [defining `_LARGEFILE64_SOURCE`](https://git.musl-libc.org/cgit/musl/tree/include/sys/stat.h?h=v1.2.5#n156). Which apparently is because [LFS64 interfaces are deprecated](https://musl.libc.org/releases.html) (see v1.2.4 release history).

musl suggests defining `_LARGEFILE64_SOURCE` in the compiler flags for a "short-term fix", which fixes the error and builds the profiler as expected.

May I make a quick PR fixing this issue? Seems to be trivial for now, since changing `stat64` usage to just `stat`, as musl recommends, probably requires discussion if it will affect other platforms (I see defines for Windows, MacOS and FreeBSD inside the `TracyFileRead.hpp` file) or keep the same behavior (I'm unfamiliar with the code base and the behavior of the `Tracy::FileRead` class).

--- a/server/TracyFileRead.hpp
+++ b/server/TracyFileRead.hpp
@@ -18,7 +18,7 @@
 #ifdef _MSC_VER
 #  define stat64 _stat64
 #endif
-#if defined __APPLE__ || defined __FreeBSD__
+#if defined __APPLE__ || defined __FreeBSD__ || (defined __linux__ && !defined __GLIBC__)
 #  define stat64 stat
 #endif
 

